File: Algorithm/Floyd Warshall/순위.swift
import Foundation

func solution(_ n:Int, _ results:[[Int]]) -> Int {
    var answer: Int = 0
    var resultInfo = [[Int]](repeating: [Int](repeating: 0, count: n + 1), count: n + 1)
    
    for res in results {
        resultInfo[res[0]][res[1]] = 1
        resultInfo[res[1]][res[0]] = -1
    }
    
    for k in 1...n {
        for a in 1...n {
            for b in 1...n {
                if resultInfo[a][k] == 1 && resultInfo[k][b] == 1 {
                    resultInfo[a][b] = 1
                } else if resultInfo[a][k] == -1 && resultInfo[k][b] == -1 {
                    resultInfo[a][b] = -1
                }
            }
        }
    }
    
    // for res in resultInfo {
    //     print(res)
    // }
    
    for a in 1...n {
        var cnt: Int = 0
        for b in 1...n {
            if resultInfo[a][b] == 0 {
                cnt += 1
            }
        }
        
        if cnt == 1 {
            answer += 1
        }
    } // for
    
    return answer
}

File: Algorithm/Floyd Warshall/순위.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<vector<int>> results) {
    int answer = 0;
    vector<vector<int>> adj(n+1, vector<int>(n+1, 0));
    
    for (const vector<int> result: results)
        adj[result[0]][result[1]] = 1;
    
    for (int k = 1; k <= n; ++k) {
        for (int i = 1; i <= n; ++i) {
            for (int j = 1; j <= n; ++j) {
                if (adj[i][k] && adj[k][j])
                    adj[i][j] = 1;
            }
        }
    }
    
    for (int i = 1; i <= n; ++i) {
        int cnt = 0;
        for (int j = 1; j <= n; ++j) 
            if (adj[i][j] || adj[j][i]) cnt++;
        
        if (cnt == n-1) answer++;
    }
    return answer;
}